SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -478,24 +484,36 @@ TArray<Query> InterestFactory::GetUserDefinedQueries(const AActor* InActor, cons
 			UserConstraint.OrConstraint.Append(FrequencyToConstraints.Value);
 		}
 
+		if (!UserConstraint.IsValid())
+		{
+			continue;
+		}
+
 		// All constraints have to be limited to the checked out levels, so create an AND constraint with the level.
 		UserQuery.Constraint.AndConstraint.Add(UserConstraint);
 		UserQuery.Constraint.AndConstraint.Add(LevelConstraint);
 
 		// We enforce result type even for user defined queries. Here we are assuming what a user wants from their defined
 		// queries are for their players to check out more actors than they normally would, so use the client non auth result type,
 		// which includes all components required for a client to see non-authoritative actors.
-		if (GetDefault<USpatialGDKSettings>()->bEnableResultTypes)
-		{
-			UserQuery.ResultComponentId = ClientNonAuthInterestResultType;
-		}
-		else
+		SetResultType(UserQuery, ClientNonAuthInterestResultType);
+		
+		AddComponentQueryPairToInterestComponent(OutInterest, SpatialConstants::GetClientAuthorityComponent(Settings->UseRPCRingBuffer()), UserQuery);
+
+		// Add the user interest to the server as well if load balancing is enabled and the client queries on server flag is flipped
+		// Need to check if load balancing is enabled otherwise there is not chance the client could see and entity the server can't,
+		// which is what the client queries on server flag is to avoid.
+		if (Settings->bEnableClientQueriesOnServer)
```

Comment:
This functionality is repeated line: 597
Could pull out a little function if you like as you did with SetResultType

Reply:
oh and 452

Reply:
If so I guess we can remove the:
const USpatialGDKSettings* Settings = GetDefault<USpatialGDKSettings>();

Reply:
The settings check is repeated, however building the query is bespoke to the different cases. How would you recommend pulling out this functionality without building the different constraints outside the flag check?

Reply:
Yes this is true for the case in line 445. Don't ignore my comments.

Reply:
😅

Reply:
Sorry I meant ignore my comments loooool

Reply:
Big MonkaS